### PR TITLE
Get it working: Correct dependencies

### PR DESCRIPTION
**Pull request by danielbeardsley**
Use the correct dependencies and drop the only `let` usage.

Seem commits for more info.

### CODE REVIEW

File: lib/authentication.js [hunk 1]
```diff
@@ -48,10 +48,9 @@ module.exports = {
                   debug('Got login from ' + user.username);
 
                   // Redirect user to original URL. Fallback to '/' if DNE.
-                  let redirectUrl = req.session.auth_redirect;
                   delete req.session.auth_redirect;
 
-                  res.redirect(redirectUrl || '/');
+                  res.redirect(req.session.auth_redirect || '/');
```
Comment (copperwall) [inline lib/authentication.js]: This uses `req.session.auth_redirect` after the field is deleted D:

Comment (danielbeardsley) [inline lib/authentication.js]: I'm a dum-dum. Thanks!

**Comment by danielbeardsley** [comment]
CR 👍

File: lib/authentication.js [hunk 2]
```diff
@@ -40,6 +40,11 @@ module.exports = {
                next(err);
             }
 
+            if (user === undefined) {
+               debug("Bad user");
+               return;
```
Comment (danielbeardsley) [inline lib/authentication.js]: I think we need to also either respond or call `next()`

File: lib/authentication.js [hunk 3]
```diff
@@ -40,6 +40,11 @@ module.exports = {
                next(err);
```
Comment (danielbeardsley) [inline lib/authentication.js]: Hmm.. shouldn't this be `return next()`? Maybe that's the source of the problem?

Comment (copperwall) [inline lib/authentication.js]: Hmm maybe? I'll check it out.

**Comment by danielbeardsley** [comment]
CR 👍

**Comment by danielbeardsley** [comment]
QA 👍

**Pull request merged**